realm/realm-library/src/main/java/io/realm/ProxyState.java

```diff
@@ -30,22 +30,28 @@
  * This implements {@code RealmObjectProxy} interface, to eliminate copying logic between
  * {@link RealmObject} and {@link DynamicRealmObject}.
  */
-public final class ProxyState {
+public final class ProxyState<E extends RealmModel> {
+    private E model;
     private String className;
     private Class<? extends RealmModel> clazzName;
 
     private Row row;
     private BaseRealm realm;
 
-    private final List<RealmChangeListener> listeners = new CopyOnWriteArrayList<RealmChangeListener>();
+    private final List<RealmChangeListener<E>> listeners = new CopyOnWriteArrayList<RealmChangeListener<E>>();
     private Future<Long> pendingQuery;
     private boolean isCompleted = false;
     protected long currentTableVersion = -1;
 
     public ProxyState() {}
 
-    public ProxyState(Class<? extends RealmModel> clazzName) {
+    public ProxyState(E model) {
```

Comment:
`DynamicRealmObject` is not really a proxy. This is used to return the `RealmObject`, `Pojo implements RealmModel` or `DynamicRealmObject` the common things I can think of is: Table/Row, Object, Model ...